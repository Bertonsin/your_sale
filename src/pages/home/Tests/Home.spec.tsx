import { describe, it, expect } from 'vitest';
import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Home from '..';

describe('home screen', () => {
  it('should display a Header and a Banner', () => {
    const { container } = render(
      <BrowserRouter>
        <Home />
      </BrowserRouter>
    );

    expect(container).matchSnapshot();
  });
});

describe('home screen buttons', () => {
  it('should be rendered on screen', () => {
    render(
      <BrowserRouter>
        <Home />
      </BrowserRouter>
    );

    expect(screen.getAllByRole('button', { name: /card/i }));
  });
});
