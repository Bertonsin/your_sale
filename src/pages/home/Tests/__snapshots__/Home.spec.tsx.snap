// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`home screen > should display a Header and a Banner 1`] = `
<div>
  <div
    class="chakra-container css-1tusmyk"
  >
    <div
      class="chakra-stack css-n07rzw"
    >
      <div
        class="chakra-stack css-7ffuev"
      >
        <div
          class="css-1tw5ajb"
        >
          <img
            class="chakra-image css-0"
            src="/Coins.png"
          />
        </div>
        <div
          class="chakra-stack css-izft5u"
        >
          <h2
            class="chakra-heading css-1p9c8n6"
          >
            <p
              class="chakra-text css-0"
            >
              Your Sale
            </p>
          </h2>
          <p
            class="chakra-text css-0"
          >
            Realize suas vendas de forma rápida e segura
          </p>
        </div>
        <div
          class="css-1tw5ajb"
        >
          <img
            class="chakra-image css-0"
            src="/Phone.png"
          />
        </div>
      </div>
      <div
        class="chakra-stack css-1yih676"
      >
        <div
          class="css-k59f1h"
        >
          <div
            class="css-128iqam"
          >
            <p
              class="chakra-text css-0"
            >
              Voce podera iniciar sua venda clicando em
              <strong>
                 NOVA VENDA 
              </strong>
              assim preenchendos os dados necessarios e gerando um novo pedido
            </p>
          </div>
          <div
            class="css-0"
          >
            <p
              class="chakra-text css-1ncuk4g"
            >
              Voce podera visualizar todas vendas criadas clicando em
              <strong>
                 LISTA DE PEDIDOS 
              </strong>
               assim podendo visualizar a listagem de pedidos e verificar detalhe do pedido criado.
            </p>
          </div>
        </div>
        <div
          class="chakra-stack css-majmq"
        >
          <button
            aria-label="card"
            class="chakra-button css-xlew0l"
            type="button"
          >
            <div
              class="chakra-stack css-k9twck"
            >
              <img
                class="chakra-image css-1t9pz9x"
                src="/newOrder.png"
              />
              <div
                class="css-1eif5ff"
              >
                <p
                  class="chakra-text css-1g6pthj"
                >
                  Nova Venda
                </p>
                <p
                  class="chakra-text css-1ncuk4g"
                >
                  Inicie uma nova venda
                </p>
              </div>
            </div>
          </button>
          <button
            aria-label="card"
            class="chakra-button css-82b7o0"
            type="button"
          >
            <div
              class="chakra-stack css-k9twck"
            >
              <img
                class="chakra-image css-1t9pz9x"
                src="/orderList.png"
              />
              <div
                class="css-1eif5ff"
              >
                <p
                  class="chakra-text css-1g6pthj"
                >
                  Lista de pedidos
                </p>
                <p
                  class="chakra-text css-1ncuk4g"
                >
                  Ver pedidos lançados
                </p>
              </div>
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
